govuk-one-login/di-account-management-backend
# OLH-1297 - Modify the format-activity-log Lambda to use the new data model

## Proposed changes

Use the new data model, such that when new activity log TXMA Events are added to the dynamo, we format correctly as per new data model before adding to the Queue for write to the activity log table.

JIRA: https://govukverify.atlassian.net/browse/OLH-1297


### What changed
Picks up TXMA event of activity log as per new model and generate an activity log that is consumable by the write activity log lambda.

### Why did it change
To align with the new data model for thew activity log.

### Related links

## Checklists

### Environment variables or secrets


### Permissions

## Testing

1. Navigate to dev environment
2. Add a new valid record to the Dynamo DB with TxMA Event structure as per new data model
3. Should add an ActivityLogEntry to the ActivityLogToWriteQueue
4. Add a new invalid record to  Dynamo DB
5. Should add an entry to the FormatActivityLogDeadLetterQueue


Sample correct new TxMA Event as per new model
`{"event_id":"12345678","timestamp":123456789,"timestamp_ms_formatted":"x","timestamp_ms":1234,"event_name":"AUTH_AUTH_CODE_ISSUED","client_id":"client-id-value","session_id":"123456789","user_id":"qwerty","reported_suspicious":true}
  `

Sample correct output ActivityLogEntry as per new model is:
`{
  "event_id": "12345678",
  "event_type": "AUTH_AUTH_CODE_ISSUED",
  "session_id": "123456789",
  "user_id": "qwerty",
  "client_id": "client-id-value",
  "timestamp": 123456789,
  "reported_suspicious": true
}`



## How to review
Code completeness  and quality, tests
